components/controller.ts

```diff
@@ -393,6 +398,10 @@ export class Controller {
      * Note: if you are adding a contract, please use {@link addMission}!
      */
     public contracts: Map<string, MissionManifest> = new Map()
+
+    // Converts a contract's repo ID to public ID.
+    public contractIdRepoToPublic: Map<string, string> = new Map()
```

Comment:
What do you mean a contracts repo ID? They don't have those.

Reply:
I mean this. I know it's called "Id". I was trying to avoid confusion with "public Id". https://github.com/thepeacockproject/Peacock/blob/df5e3ae373f326317a89f071a98794c4d5732f56/contractdata/FEATURED/fortheseven-FC1.json#L299

Reply:
But this causes more confusion, as contracts are not part of the REPO.

Reply:
Additionally, this is just redundant, as we can do `resolveContract(id)?.Metadata.PublicId`.

Reply:
Naming changed.

`resolveContract` doesn't work because it relies on having the `json` file of the contract locally, but here the public ID is needed to download the contract, so we don't have them locally yet.